addons/test_mail/tests/test_ir_actions.py

```diff
@@ -181,6 +185,167 @@ def test_action_next_activity_from_x2m_user(self):
+
+            action_form.activity_type_id = self.env.ref('mail.mail_activity_data_todo')
+            self.assertEqual(action_form.name, 'Create To-Do')
+            self.assertEqual(action_form.activity_summary, 'To-Do')
+            self.assertEqual(action_form.activity_date_deadline_range, 1)
+            self.assertEqual(action_form.activity_date_deadline_range_type, 'days')
+            self.assertEqual(action_form.activity_user_type, 'specific')
+            self.assertFalse(action_form.activity_user_id)
+            self.assertFalse(action_form.activity_note)
+
+            action_form.activity_summary = 'Send a wonderful email'
+            action_form.activity_note = 'Hello world'
+            action_form.activity_date_deadline_range = 3
+            action_form.activity_date_deadline_range_type = 'weeks'
+            action_form.activity_user_id = self.user_demo
+            action_form.activity_type_id = email_activity_type
+            self.assertEqual(action_form.activity_summary, 'Email', 'activity_summary should be changed to "Email"')
+            self.assertEqual(action_form.activity_note, '<p>Default note for email</p>', 'activity_note should be changed to default note of email activity')
+            self.assertEqual(action_form.activity_date_deadline_range, 3)
+            self.assertEqual(action_form.activity_date_deadline_range_type, 'weeks')
+            self.assertEqual(action_form.activity_type_id, email_activity_type)
+            self.assertEqual(action_form.activity_user_type, 'specific')
+            self.assertEqual(action_form.activity_user_id, self.user_demo)
+
+            action_form.activity_user_type = 'generic'
+            self.assertFalse(action_form.activity_user_id)
+            self.assertEqual(action_form.activity_user_field_name, 'user_id')
+
+            action_form.activity_type_id = self.env.ref('mail.mail_activity_data_call')
+            self.assertEqual(action_form.activity_type_id, self.env.ref('mail.mail_activity_data_call'))
+            self.assertEqual(action_form.activity_user_type, 'generic')
+            self.assertEqual(action_form.activity_user_field_name, 'user_id')
+            self.assertEqual(action_form.activity_summary, 'Call')
+            self.assertFalse(action_form.activity_note, 'activity_note should be changed to default note of call activity (which is empty)')
+            self.assertEqual(action_form.activity_date_deadline_range, 3)
+            self.assertEqual(action_form.activity_date_deadline_range_type, 'weeks')
+
+    def test_action_next_activity_formview_with_plans(self):
+        """ Test the form view's onchanges/computed of next activity server actions. Specifically when activity plans are involved. """
+        activity_type_todo = self.env.ref('mail.mail_activity_data_todo')
+        admin_activity_plan = self.env['mail.activity.plan'].create({
+            'name': 'Test Onboarding Plan (responsible: admin)',
+            'res_model': 'res.partner',
```

Comment:
You may use a model from test_mail (allows to be independent from partner specific access rights, definitions, field names, ...). Several of them are inheriting from activity mixin. Not mandatory, just that we try to be as independent from business models as possible when we test generic mixins.